src/xtgeo/surface/_regsurf_import.py

```diff
@@ -95,78 +90,69 @@ def _import_irap_binary_purepy(self, mfile, values=True):
             break
 
     values = np.hstack(datav)
-    values = np.reshape(values, (self._ncol, self._nrow), order="F")
+    values = np.reshape(values, (args["ncol"], args["nrow"]), order="F")
     values = np.array(values, order="C")
     values = np.ma.masked_greater_equal(values, UNDEF_MAP_IRAPB)
-    self._values = np.ma.masked_invalid(values)
-
-    self._ilines = np.array(range(1, self.ncol + 1), dtype=np.int32)
-    self._xlines = np.array(range(1, self.nrow + 1), dtype=np.int32)
+    args["values"] = np.ma.masked_invalid(values)
 
     del buf
+    return args
 
 
-def _import_irap_binary(self, mfile, values=True):
+def _import_irap_binary(mfile, values=True):
 
     logger.info("Enter function %s", __name__)
 
     cfhandle = mfile.get_cfhandle()
-
+    args = {}
     # read with mode 0, to get mx my and other metadata
     (
         ier,
-        self._ncol,
-        self._nrow,
+        args["ncol"],
+        args["nrow"],
         _,
-        self._xori,
-        self._yori,
-        self._xinc,
-        self._yinc,
-        self._rotation,
+        args["xori"],
+        args["yori"],
+        args["xinc"],
+        args["yinc"],
+        args["rotation"],
         val,
     ) = _cxtgeo.surf_import_irap_bin(cfhandle, 0, 1, 0)
 
     if ier != 0:
         mfile.cfclose()
         raise RuntimeError("Error in reading Irap binary file")
 
-    self._yflip = 1
-    if self._yinc < 0.0:
-        self._yinc *= -1
-        self._yflip = -1
+    args["yflip"] = 1
+    if args["yinc"] < 0.0:
+        args["yinc"] *= -1
+        args["yflip"] = -1
 
     # lazy loading, not reading the arrays
-    if not values:
-        self._values = None
-        mfile.cfclose()
-        return
-
-    nval = self._ncol * self._nrow
-    xlist = _cxtgeo.surf_import_irap_bin(cfhandle, 1, nval, 0)
-    if xlist[0] != 0:
-        mfile.cfclose()
-        raise RuntimeError("Problem in {}, code {}".format(__name__, ier))
-
-    val = xlist[-1]
+    if values:
+        nval = args["ncol"] * args["nrow"]
+        xlist = _cxtgeo.surf_import_irap_bin(cfhandle, 1, nval, 0)
+        if xlist[0] != 0:
+            mfile.cfclose()
+            raise RuntimeError("Problem in {}, code {}".format(__name__, ier))
 
-    val = np.reshape(val, (self._ncol, self._nrow), order="C")
+        val = xlist[-1]
```

Comment:
Does this have a test? I would expect this to fail for `values=False`.. `xlist` is otherwise undefined.

Reply:
It is inside a check making sure that ` values` is truthy

Reply:
Not sure how, but somehow I read this as the `val = xlist[-1]` being an indentation back..